UI/FleetButton.cpp

```diff
@@ -172,13 +172,28 @@ FleetButton::FleetButton(const std::vector<int>& fleet_IDs, SizeType size_type)
 
     // select icon(s) for fleet(s)
     int num_ships = 0;
+    m_fleet_blockaded = false;
     for (std::shared_ptr<const Fleet> fleet : fleets) {
-        if (fleet)
+        if (fleet) {
             num_ships += fleet->NumShips();
+            if (fleet->Blockaded())
+                m_fleet_blockaded = true;
```

Comment:
can presumably break out of the loop in the event a fleet is found, as `m_fleet_blockaded` can't get any more or less `true` than it is made here...